pt/riscv_test.h

```diff
@@ -5,15 +5,26 @@
 
 #include "../p/riscv_test.h"
 
-#define TIMER_INTERVAL 2
+#define TIMER_INTERVAL 50
+#define MTIME_BASE       0x0200BFF8
+#define MTIMECMP_BASE    0x02004000
+#define MTIMECMP_SHIFT   3  // 64 bits for rv32/64 
 
 #undef EXTRA_INIT_TIMER
 #define EXTRA_INIT_TIMER                                                \
         li a0, MIP_MTIP;                                                \
         csrs mie, a0;                                                   \
-        csrr a0, mtime;                                                 \
-        addi a0, a0, TIMER_INTERVAL;                                    \
-        csrw mtimecmp, a0;                                              \
+	csrr t0, mstatus;						\
+	ori t0, t0, MSTATUS_MPIE;					\
```

Comment:
Why are you changing PIE here?

Reply:
on mret. mie in mstatus will become mpie in mstatus. mie in mstatus is the global interrupt enable signal.

Reply:
That’s done in hardware

Reply:
Yes. If I do not write MPIE before mret, MIE will be clearer after mret.

Reply:
No it won’t. On the timer interrupt, MPIE becomes MIE (1) in hardware. On MRET, MIE becomes MPIE (1)

Reply:
Because on MRET, MIE becomes MPIE. I write MPIE to 1 to before MRET to enable timer interrupt after MRE.

Reply:
You’re not understanding. MPIE is already 1 when you’re in the trap. If it weren’t, you wouldn’t have taken the trap

Reply:
Look at how the old spike code worked. They didn’t need to write to MPIE

Reply:
This MPIE is only for initialization. During reset mstatus was all zeros.